app/services/payment_registration_guard.rb

```diff
@@ -17,12 +17,19 @@ def requires_registration_before_posting?
   private
 
   def requires_registration?
```

Comment:
Should query from shape when creating a new listing and from listing when buying.